packages/react-utilities/src/hooks/useOnClickOutside.ts

```diff
@@ -1,58 +1,11 @@
-import * as React from 'react';
-import { useEventCallback } from './useEventCallback';
+import { useOnEventOutside, UseOnEventOutsideOptions } from './useOnEventOutside';
 
-export type UseOnClickOutsideOptions = {
-  /**
-   * The element to listen for the click event
-   */
-  element: Document | undefined;
-  /**
-   * Refs to elements that check if the click is outside
-   */
-  refs: React.MutableRefObject<HTMLElement | undefined | null>[];
-  /**
-   * Called if the click is outside the element refs
-   */
-  callback: (ev: MouseEvent | TouchEvent) => void;
-
-  /**
-   * By default uses element.contains, but custom contain function can be provided
-   * @param parentRef - provided parent ref
-   * @param child - event target element
-   */
-  contains?(parent: HTMLElement | null, child: HTMLElement): boolean;
-
-  /**
-   * Disables event listeners
-   */
-  disabled?: boolean;
-};
+export type UseOnClickOutsideOptions = UseOnEventOutsideOptions<MouseEvent | TouchEvent>;
 
 /**
  * Utility to perform checks where a click/touch event was made outside a compoent
  */
 export const useOnClickOutside = (options: UseOnClickOutsideOptions) => {
-  const { refs, callback, element, disabled, contains: containsProp } = options;
-
-  const listener = useEventCallback((ev: MouseEvent | TouchEvent) => {
-    const contains: UseOnClickOutsideOptions['contains'] =
-      containsProp || ((parent, child) => !!parent?.contains(child));
-
-    const isOutside = refs.every(ref => !contains(ref.current || null, ev.target as HTMLElement));
-    if (isOutside && !disabled) {
-      callback(ev);
-    }
-  });
-
-  React.useEffect(() => {
-    if (!disabled) {
-      element?.addEventListener('click', listener);
-      element?.addEventListener('touchstart', listener);
-    }
-
-    return () => {
-      element?.removeEventListener('click', listener);
-      element?.removeEventListener('touchstart', listener);
-    };
-  }, [listener, element, disabled]);
+  useOnEventOutside<MouseEvent>({ ...options, event: 'click' });
+  useOnEventOutside<MouseEvent>({ ...options, event: 'touchstart' });
```

Comment:
I am not 100% that it's a good idea:
- it doubles amount of hooks for no reason
  - before: `useEventCallback()` & `useEffect()`
  - after: (`useEventCallback()` & `useEffect()`) * 2
- there is a problem with React 17 that requires a fix only for "click" events (as I remember)

https://github.com/microsoft/fluentui/blob/d2504c72ec7fba0cf4dcf1984c2e1a9322720225/packages/fluentui/react-component-event-listener/src/useEventListener.ts#L52-L55

Reply:
I have 2 alternatives then:
* Accept array of events (I prefer this one)
* Duplicate the code between `useOnClickOutside` and `onScrollOutside`

WDYT ?

Reply:
> * Duplicate the code between `useOnClickOutside` and `onScrollOutside`

I would duplicate for now as we will need special handling for a click outside.

Reply:
Duplicated, at least I deduped the typings